packages/sanity/src/core/form/studio/tree-editing/utils/build-tree-editing-state/utils.ts

```diff
@@ -0,0 +1,34 @@
+import {type Path} from 'sanity'
+
+/**
+ * Check if the item is selected
+ */
+export function isSelected(itemPath: Path, path: Path): boolean {
+  return JSON.stringify(itemPath) === JSON.stringify(path)
+}
+
+/**
+ * Check if the path is an array item path
+ */
+export function isArrayItemPath(path: Path): boolean {
+  return path[path.length - 1].hasOwnProperty('_key')
```

Comment:
Could you add a comment on why you need to do `-1`?

Also what would happen if the path.length is `0` this will cause a typeerror? Should a check for that be added?

Reply:
Technically I don't think it's possible to it ever be 0 since we're always a level down and so never at "document level" that being said I think it's wise to return the method before just in case if this method is by accident called at doc level

Reply:
Anyway, I've added a guardrail for it :)